kinesumer.go

```diff
@@ -488,30 +494,37 @@ func (k *Kinesumer) consumePipe(stream string, shard *Shard) {
 				}
 			}
 		case <-checkPointTicker.C:
-			k.commitCheckPoint(stream, shard.ID, lastSequence)
+			k.Commit(stream, shard.ID, lastSequence)
 		}
 	}
 }
 
-// update checkpoint using sequence number.
-func (k *Kinesumer) commitCheckPoint(stream, shardID, lastSeqNum string) {
-	if lastSeqNum == "" {
+// Commit updates check point with sequence number.
+func (k *Kinesumer) Commit(stream, shardID, sequenceNumber string) {
```

Comment:
In general, clients don't know about record metadata such as stream, shard id (clients don't care about where is a record from, and clients don't need to know it), and internal sequence number. So this API interface looks not client-friendly. 

Passing the record metadata to commit should be handled by the Kinesumer, not by clients. Kinesumer can extract the metadata from the record, so we can just pass the received record back to this Commit function. (Record struct has no metadata about shard id, we may need to add it.)